### PR TITLE
Adding support for cucumber expressions such as {word}

### PR DESCRIPTION
**Pull request by JornWildt**
This PR adds support for cucumber expressions as described in https://github.com/cucumber/cucumber-expressions#readme

### CODE REVIEW

**Comment by JornWildt** [comment]
Fixed wrong method call.

File: source/Xunit.Gherkin.Quick/CoreModel/StepMethod/ScenarioStepPattern.cs [hunk 1]
```diff
@@ -7,18 +7,39 @@ namespace Xunit.Gherkin.Quick
     internal sealed class ScenarioStepPattern
     {
         private readonly string _pattern;
-        public string Pattern { get { return _pattern; } }
 
+        /// <summary>
+        /// Original pattern as supplied from step attributes.
+        /// </summary>
+        public string Pattern { get { return _pattern; } }
```
Comment (ttutisani) [inline source/Xunit.Gherkin.Quick/CoreModel/StepMethod/ScenarioStepPattern.cs:14]: If you added RegexPattern, please rename this to OriginalPattern for clarity. Comment helps, but the name can be also improved.

**Review by ttutisani — COMMENTED** [review]
This is good work! Please address my comments and I will review them again.
Thanks for your time!

File: source/Xunit.Gherkin.Quick.ProjectConsumer/CucumberExpressions/PassParameters.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        [When(@"I choose the integer value {int}")]
+        public void SetInteger(int i)
+        {
+            IntegerValue = i;
+        }
+
+        [Then(@"the triple value is {int}")]
+        public void CheckTriple(int t)
+        {
+            Assert.Equal(t, IntegerValue * 3);
```
[the remark this answers is not in the document]
Comment (JornWildt) [reply source/Xunit.Gherkin.Quick.ProjectConsumer/CucumberExpressions/PassParameters.cs:28]: Done.

File: source/Xunit.Gherkin.Quick/CoreModel/StepMethod/ScenarioStepPattern.cs [hunk 3]
```diff
@@ -7,18 +7,39 @@ namespace Xunit.Gherkin.Quick
...
+        public string RegexPattern { get { return _regexPattern; } }
+
         public PatternKind Kind { get; }
 
         public ScenarioStepPattern(string pattern, PatternKind stepMethodKind)
         {
             _pattern = !string.IsNullOrWhiteSpace(pattern) 
                 ? pattern 
                 : throw new ArgumentNullException(nameof(pattern));
+            _regexPattern = ConvertSpecialPatternsToRegex(_pattern);
```
[the remark this answers is not in the document]
Comment (JornWildt) [reply source/Xunit.Gherkin.Quick/CoreModel/StepMethod/ScenarioStepPattern.cs:29]: Done

File: source/Xunit.Gherkin.Quick/CoreModel/StepMethod/ScenarioStepPattern.cs (hunk 1 above)
Comment (JornWildt) [reply source/Xunit.Gherkin.Quick/CoreModel/StepMethod/ScenarioStepPattern.cs:14]: Done

**Pull request merged**